node/src/actors/chain_manager/mod.rs

```diff
@@ -542,26 +542,17 @@ impl ReputationInfo {
     ) {
         let dr_pointer = tally_transaction.dr_pointer;
         let dr_state = &data_request_pool.data_request_pool[&dr_pointer];
-        let reveals = &dr_state.info.reveals;
         let commits = &dr_state.info.commits;
         let replication_factor = dr_state.data_request.witnesses;
         self.alpha_diff += Alpha(u32::from(replication_factor));
 
-        let _tally = &tally_transaction.tally;
-
+        let mut tally_vts = HashSet::new();
+        // TODO: Review case of data requester is also a witness
+        for vt in &tally_transaction.outputs {
+            tally_vts.insert(vt.pkh);
+        }
         for pkh in commits.keys() {
-            let liar = reveals
-                .get(&pkh)
-                .map(|_reveal_tx| {
-                    // FIXME: restore this clause once `true_revealer` is unmocked.
-                    /*if true_revealer(reveal_tx, tally) {
-                        0
-                    } else {
-                        1
-                    }*/
-                    0
-                })
-                .unwrap_or(1);
+            let liar = if tally_vts.contains(pkh) { 0 } else { 1 };
```

Comment:
Add comment:
```suggestion
            // If the identity behaved honestly, it must have received a reward.
            // So we know which identities lied because they do not have a vtt transaction in the tally.
            let liar = if tally_vts.contains(pkh) { 0 } else { 1 };
```